site/app/controllers/grading/SimpleGraderController.php

```diff
@@ -131,4 +134,80 @@ public function save($action) {
         $this->core->getOutput()->renderJson($response);
         return $response;
     }
+
+    public function csv($action) {
+
+        $users = $_POST['users'];
+        $g_id = $_POST['g_id'];
+        $num_numeric = $_POST['num_numeric'];
+        $num_text = $_POST['num_text'];
+        $component_ids = $_POST['component_ids'];
+        $csv_array = preg_split("/\r\n|\n|\r/", $_POST['big_file']);
+        $arr_length = count($csv_array);
+        $return_data = array();
+
+        $data_array = array();
+        for ($i = 0; $i < $arr_length; $i++) {
+            $temp_array = explode(',', $csv_array[$i]);
+            $data_array[] = $temp_array;
+        }
+
+        foreach($users as $username) {
+            for ($j = 0; $j < $arr_length; $j++) {
+                if($username === $data_array[$j][0]) {
+                    $temp_array = array();
+                    $temp_array['username'] = $username;
+                    $num_questions = 0;
```

Comment:
Is `$num_questions` used anywhere else? If not, you should get rid of this line.